routes/location.js

```diff
@@ -35,24 +37,29 @@ function getCloudMetadata(callback) {
                     // Try Azure next
                     getAzureCloudMetadata(function(err, c, z) {
                         if (err) {
-                            // Try GCP next
                             getGCPCloudMetadata(function(err, c, z) {
-                                // Return result regardless of error
-                                callback(c, z); // Running in GCP or unknown
+                                if (err) {
+                                    // Try GCP next
```

Comment:
This comment should remain for the `getGCPCloudMetadata` invocation and a new comment describing what `getGeneric` is intended for.